dbms/src/Storages/Page/V3/PageDirectory.cpp

```diff
@@ -1484,6 +1491,8 @@ std::unordered_set<String> PageDirectory<Trait>::apply(PageEntriesEdit && edit,
     watch.restart();
 
     writers.push_back(&w);
+    SYNC_FOR("after_PageDirectory::enter_write_group");
+    CurrentMetrics::set(CurrentMetrics::PSWriterQueueSize, writers.size());
```

Comment:
when there are multiple PS instances(deployed under non-disagg mode), this will randomly show the queue size of one of the PS instance. 🤔️

Reply:
Yes, it also has other problems. Change to `CurrentMetrics::Increment pending_writer_size{CurrentMetrics::PSPendingWriterSize};`